app/src/androidTest/java/com/android/periodpals/endtoend/EndToEndM2.kt

```diff
@@ -45,60 +47,129 @@ class EndToEndM2 : TestCase() {
     composeTestRule.setContent { MainActivity() }
   }
 
+  /**
+   * End-to-end test for the
+   * [edit profile flow](https://www.figma.com/design/r6jgyWnwTQ6e5X1eLpeHwN/PeriodsPals?node-id=579-5989&node-type=canvas&m=dev)
+   *
+   * The "user" lands on the SignIn screen then signs in to their existing account. They click on
+   * the edit button in the top app bar and gets redirected to the EditProfile screen. They
+   * (correctly) fill in the fields and click on the "Save" button and get redirected to the Profile
+   * screen that displays the info they just entered.
+   */
   @Test
-  fun signInAndEditProfile() {
-    // Sign in using existing account
+  fun editProfileEndToEnd() {
+
+    // SignIn Screen
+    // User signs in using existing account
     composeTestRule.waitForIdle()
     Log.d(TAG, "User arrives on SignIn Screen")
-    composeTestRule.onNodeWithTag(SignInScreen.SCREEN).assertExists() // or is displayed?
+    composeTestRule.onNodeWithTag(SignInScreen.SCREEN).assertExists()
     composeTestRule
         .onNodeWithTag(AuthenticationScreens.EMAIL_FIELD)
+        .performScrollTo()
         .assertIsDisplayed()
         .performTextInput(EMAIL)
     composeTestRule
         .onNodeWithTag(AuthenticationScreens.PASSWORD_FIELD)
+        .performScrollTo()
         .assertIsDisplayed()
         .performTextInput(PASSWORD)
-    composeTestRule.onNodeWithTag(SignInScreen.SIGN_IN_BUTTON).assertIsDisplayed().performClick()
+    composeTestRule
+        .onNodeWithTag(SignInScreen.SIGN_IN_BUTTON)
+        .performScrollTo()
+        .assertIsDisplayed()
+        .performClick()
     composeTestRule.waitUntil(TIMEOUT) {
       composeTestRule.onAllNodesWithTag(ProfileScreen.SCREEN).fetchSemanticsNodes().size == 1
     }
 
-    // Profile Screen is displayed
+    // Profile Screen
+    // User arrives on their profile
     composeTestRule.waitForIdle()
     Log.d(TAG, "User arrives on Profile Screen")
     composeTestRule.onNodeWithTag(TopAppBar.EDIT_BUTTON).assertIsDisplayed().performClick()
     composeTestRule.waitUntil(TIMEOUT) {
       composeTestRule.onAllNodesWithTag(EditProfileScreen.SCREEN).fetchSemanticsNodes().size == 1
     }
 
-    // Edit Profile Screen is displayed, edit name, dob and description
+    // Edit Profile Screen
+    // User edits their profile
     composeTestRule.waitForIdle()
     Log.d(TAG, "User arrives on Edit Profile Screen")
     composeTestRule
         .onNodeWithTag(ProfileScreens.NAME_INPUT_FIELD)
+        .performScrollTo()
+        .assertIsDisplayed()
+        .performTextClearance()
+    composeTestRule
+        .onNodeWithTag(ProfileScreens.NAME_INPUT_FIELD)
+        .performScrollTo()
         .assertIsDisplayed()
         .performTextInput(name)
```

Comment:
Good point to add the scrolls and text clearance.  
However, I wonder if it is necessary to `composeTestRule(Tag).perfromScrollTo().assertIsDisplayed()` twice here with the first a text clearance and the second a text input.
```suggestion
    composeTestRule
        .onNodeWithTag(ProfileScreens.NAME_INPUT_FIELD)
        .performScrollTo()
        .assertIsDisplayed()
        .performTextClearance()
        .performTextInput(name)
```

Same remark for other instances.

Reply:
I wondered as well, but it turns out `performTextClearance` returns a `Unit` and not the `SemanticsNodeInteraction` we would need to chain these calls :(

Reply:
Too bad then... :(